common/conf.hpp

```diff
@@ -92,9 +92,128 @@ class CmdlineOptionsIterator {
 	ArgumentsMode mode_ {ArgumentsMode::RejectBareArguments};
 };
 
+// NOTE - When updating this class - either adding or removing variables. Be
+// sure to update the transitive dependencies also.
+class DefaultConf {
+private:
+	string DefaultPathConfDir = conf::GetEnv("MENDER_CONF_DIR", path::Join("/etc", "mender"));
```

Comment:
4 things here, all YMMV-style, but I'd:
- call this `Paths`
- make it a `struct`
- not call things `Default` when they actually hold the current value that no longer necessarily matches the actual/original default
- move this into `MenderConfig` because there should not be any (other) instances outside of it

Reply:
Agree with 1 and 3. I don't think it should be a struct because some of the setters are doing non-trivial things, and all the members are private for that reason.

Number 4.. I mean we could, but I don't see high value in it, and I suppose keeping it outside is easier to test. The namespace isn't too crowded anyway.